libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +225,68 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+async fn bip353_parse(
+    input: &str,
+    dns_resolver: &AsyncResolver<GenericConnector<TokioRuntimeProvider>>,
+) -> Option<String> {
+    if let Some((local_part, domain)) = input.split_once('@') {
+        let dns_name = format!("{}.{}.{}", local_part, USER_BITCOIN_PAYMENT_PREFIX, domain);
+
+        // Query for TXT records of a domain
+        let txt_data = match dns_resolver.txt_lookup(dns_name).await {
+            Ok(records) => records
+                .iter()
+                .flat_map(|record| record.to_string().into_bytes())
+                .collect::<Vec<u8>>(),
+            Err(e) => {
+                debug!("No BIP353 TXT records found: {}", e);
+                return None;
+            }
+        };
+
+        // Decode TXT data
+        match String::from_utf8(txt_data) {
+            Ok(decoded) => {
+                // FIXME: The idea is to return from here the complete uri and parse it inside the parse_core function
+                // return Some(decoded);
+
+                #[cfg(feature = "liquid")]
+                if let Some(bolt12_offer) = extract_bolt12_offer(&decoded) {
+                    return Some(bolt12_offer.to_string());
+                }
+
+                if let Some(lnurl) = extract_lnurl(&decoded) {
+                    return Some(lnurl.to_string());
+                }
```

Comment:
This doesn't take into account that there might be two query params one for offer and one for lnurl for example:
`bitcoin:?lno=<offer>&lnurl=<lnurl>`
I suggest:
1. First validate and parse the url which should be ended with the query params parsed in hand.
2. Then look according to priorities for the lno in the parsed params and then for the lnurl.